packages/axelar-wasm-std/src/msg_id/aleo_message_id.rs

```diff
@@ -0,0 +1,64 @@
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+use aleo_types::{Transaction, Transition};
+use error_stack::Report;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use super::Error;
+
+pub struct AleoMessageId {
+    pub transaction_id: Transaction,
+    pub transition_id: Transition,
+}
+
+impl AleoMessageId {
+    pub fn new(transaction_id: Transaction, transition_id: Transition) -> Self {
+        Self {
+            transaction_id,
+            transition_id,
+        }
+    }
+}
+
+const PATTERN: &str =
+    "^(at1[qpzry9x8gf2tvdw0s3jn54khce6mua7l]{58})-(au1[qpzry9x8gf2tvdw0s3jn54khce6mua7l]{58})$";
```

Comment:
We can put the in alphanumeric order if you prefer it.
The order follows the order of "Bech32"
https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#bech32

I have copied the table here for reference:

|     | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 |
|-----|---|---|---|---|---|---|---|---|
| +0  | p | q | z | r | y | 9 | x | 8 |
| +8  | g | f | 2 | t | v | d | w | 0 |
| +16 | s | 3 | j | n | 5 | 4 | k | h |
| +24 | c | e | 6 | m | u | a | 7 | l |

Just let me know how you prefer it

Reply:
I would prefer to make the pattern as readable as possible, otherwise regex is easy to get wrong. You could also consider using ranges to make it more obvious which letters are skipped. I think `[02-9a-hj-np-z]` works, but please double-check